app/controllers/early_allocations_controller.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+class EarlyAllocationsController < PrisonsApplicationController
+  def new
+    @offender = EarlyAllocation.new offender_id_from_url
+  end
+
+  def create
+    @offender = EarlyAllocation.new early_allocation_params.merge(offender_id_from_url)
+    if @offender.stage1_complete?
+      stage2
+    else
+      stage1
+    end
+  end
+
+  def show
+    @offender = EarlyAllocation.find_by! offender_id_from_url
+    @prisoner = OffenderService.get_offender(params[:offender_id])
+    @allocation = AllocationVersion.find_by!(offender_id_from_url)
+    @pom = PrisonOffenderManagerService.get_pom(@prison, @allocation.primary_pom_nomis_id)
+
+    respond_to do |format|
```

Comment:
If this is still pending (I think there's another ticket to be able to mark it as complete), should we show stage2 for the html view?